src/attribution/onINP.ts

```diff
@@ -260,6 +261,96 @@ export const onINP = (
+    if (!attribution.longAnimationFrameEntries?.length) {
+      return;
+    }
+
+    const interactionTime = attribution.interactionTime;
+    const inputDelay = attribution.inputDelay;
+    const processingDuration = attribution.processingDuration;
+
+    // Stats across all LoAF entries and scripts.
+    let totalScriptDuration = 0;
+    let totalStyleAndLayoutDuration = 0;
+    let totalPaintDuration = 0;
+    let longestScriptDuration = 0;
+    let longestScriptEntry: PerformanceScriptTiming | undefined;
+    let longestScriptSubpart: INPSubpart | undefined;
+
+    for (const loafEntry of attribution.longAnimationFrameEntries) {
+      totalStyleAndLayoutDuration =
+        totalStyleAndLayoutDuration +
+        loafEntry.startTime +
+        loafEntry.duration -
+        loafEntry.styleAndLayoutStart;
+
+      for (const script of loafEntry.scripts) {
+        const scriptEndTime = script.startTime + script.duration;
+        if (scriptEndTime < interactionTime) {
+          continue;
+        }
+        const intersectingScriptDuration =
+          scriptEndTime - Math.max(interactionTime, script.startTime);
+        // Since forcedStyleAndLayoutDuration doesn't provide timestamps, we
+        // apportion the total based on the intersectingScriptDuration. Not
+        // correct depending on when it occured, but the best we can do.
+        const intersectingForceStyleAndLayoutDuration = script.duration
+          ? (intersectingScriptDuration / script.duration) *
+            script.forcedStyleAndLayoutDuration
+          : 0;
+        // For scripts we exclude forcedStyleAndLayout (same as DevTools does
+        // in it's summary totals) and instead include that in
+        // totalStyleAndLayoutDuration
+        totalScriptDuration +=
+          intersectingScriptDuration - intersectingForceStyleAndLayoutDuration;
+        totalStyleAndLayoutDuration += intersectingForceStyleAndLayoutDuration;
+
+        if (intersectingScriptDuration > longestScriptDuration) {
+          // Set the subpart this occured in.
+          longestScriptSubpart =
+            script.startTime < interactionTime + inputDelay
+              ? 'input-delay'
+              : script.startTime >=
+                  interactionTime + inputDelay + processingDuration
+                ? 'presentation-delay'
+                : 'processing-duration';
+
+          longestScriptEntry = script;
+          longestScriptDuration = intersectingScriptDuration;
+        }
+      }
+    }
+
+    // Calculate the totalPaintDuration from the last LoAF after
+    // presentationDelay starts (where available)
+    const lastLoAF = attribution.longAnimationFrameEntries.at(-1);
+    const lastLoAFEndTime = lastLoAF
+      ? lastLoAF.startTime + lastLoAF.duration
+      : 0;
+    if (lastLoAFEndTime >= interactionTime + inputDelay + processingDuration) {
+      totalPaintDuration =
+        attribution.interactionTime + value - lastLoAFEndTime;
```

Comment:
Should we use `value` here is should we use `nextPaintTime` - `startTime`, which would more accurately match the duration for `presentationDelay`?

Reply:
Yes. Fixed. Though in this case we want the time not the duration so it's even easier.
Also saves me passing `value` to this function.